### PR TITLE
Add Google compute engine support

### PR DESCRIPTION
**Pull request by stgleb**
Add support of Google Compute Engine 

API changes

          Cloud acccount format

 ```
{
    "name": "gce",
    "provider": "gce",
    "credentials": {
  "projectId": "project ID from service account",
  "privateKey": "-----BEGIN PRIVATE KEY----- BlagBlah-----END PRIVATE KEY-----\n",
  "clientEmail": "client email from service account",
  "tokenURI": "https://oauth2.googleapis.com/token",
"publicKey": "public Key goes here"
    }
}
 ```   

Provision request:

Change:

Add zone field to profile

```
{
    "clusterName": "gleb",
    "profile": {
        "masterProfiles": [
          {
                "imageFamily": "ubuntu-1804-lts",
                "size": "n1-standard-2"}
            ],
        "nodesProfiles": [
            {
                "imageFamily": "ubuntu-1804-lts",
                "size": "n1-standard-2"
            }
        ],
        "provider": "gce",
        "region": "europe-west1",
        // New Field
        "zone": "europe-west1-b",

        "arch": "amd64",
        "operatingSystem": "linux",
        "ubuntuVersion": "xenial",
        "dockerVersion": "17.06.0",
        "K8SVersion": "1.11.1",
        "flannelVersion": "0.10.0",
        "networkType": "vxlan",
        "cidr": "10.2.0.0/16",
        "helmVersion": "2.11.0",
        "rbacEnabled": true
    },
    "cloudAccountName": "gce"
}
```

Add node to existing cluster stays the same

```
[
            {
                "imageFamily": "ubuntu-1804-lts",
                "size": "n1-standard-2"
            }
        ]
```

### CODE REVIEW

File: pkg/workflows/steps/config.go [hunk 1]
```diff
@@ -44,7 +44,18 @@ type DOConfig struct {
 
 // TODO(stgleb): Fill struct with fields when provisioning on other providers is done
 
-type GCEConfig struct{}
+type GCEConfig struct {
+	PrivateKey       string `json:"privateKey"`
+	ImageFamily      string `json:"imageFamily"`
+	ProjectID        string `json:"projectId"`
```
Comment (saheienko) [inline pkg/workflows/steps/config.go]: Please, use `projectID` as a tag name.

File: pkg/account/region_finder.go [hunk 2]
```diff
@@ -45,34 +50,67 @@ type RegionSizes struct {
...
+	//GetRegions returns a slice of cloud specific regions/az's
 	//if not found would return an empty slice
-	Find(context.Context) (*RegionSizes, error)
+	GetRegions(context.Context) (*RegionSizes, error)
 }
 
-//GetRegionFinder returns finder attached to corresponding account as it has all credentials for a cloud provider
-func GetRegionFinder(account *model.CloudAccount) (RegionFinder, error) {
+//GetRegionsGetter returns finder attached to corresponding account as it has all credentials for a cloud provider
+func GetRegionsGetter(account *model.CloudAccount, config *steps.Config) (RegionsGetter, error) {
```
Comment (saheienko) [inline pkg/account/region_finder.go]: Could it be `NewRegionsGetter`?

File: pkg/kube/handler.go [hunk 3]
```diff
@@ -599,6 +610,14 @@ func (h *Handler) deleteNode(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	logrus.Println("Masters")
+	for master := range k.Masters {
+		logrus.Println(master)
```
[the remark this answers is not in the document]
Comment (stgleb) [reply pkg/kube/handler.go]: I've gorgot to delete it, only debugging purpose

File: pkg/workflows/steps/config.go (hunk 1 above)
Comment (stgleb) [reply pkg/workflows/steps/config.go]: Well, we never use ID in tag names, only in field names. This looks more conformant to js and json.

File: pkg/account/region_finder.go (hunk 2 above)
Comment (stgleb) [reply pkg/account/region_finder.go]: Yep)

File: pkg/workflows/steps/config.go (hunk 1 above)
Comment (saheienko) [reply pkg/workflows/steps/config.go]: Got it.

**Pull request merged**